components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java

```diff
@@ -160,6 +166,44 @@ public String getServiceProviderNameByClientId(String clientId, String type, Str
         return appName;
     }
 
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (StringUtils.isEmpty(inboundKey)) {
+            return null;
+        }
```

Comment:
Shall we do a null check for tenant domain and inboundType as well at the top and return